### PR TITLE
Get multiple search results

### PR DESCRIPTION
**Pull request by phlpphcknr**
- Rename various classes
- Create backend classes
- Adjust tests
- Show 5 instead of 1 artist search result 
- write method to get releases from discogs api

### CODE REVIEW

File: backend/src/main/java/com/hackner/musicompass/controller/ArtistSearchController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+public class ArtistSearchController {
+
+    private final ArtistSearchService artistSearchService;
+
+    @Autowired
+    public ArtistSearchController(ArtistSearchService artistSearchService) {
+        this.artistSearchService = artistSearchService;
+    }
+
+    @GetMapping("{searchTerm}")
```
Comment (fabianschmauder) [inline backend/src/main/java/com/hackner/musicompass/controller/ArtistSearchController.java:20]: i would change the endpoint to "api/artist" and have query parameter search or q so an url would look like
"api/artist?q=ACDC"

File: backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsArtistSearchResults.java [hunk 2]
```diff
@@ -15,6 +16,5 @@
 public class DiscogsArtistSearchResults {
 
     @JsonProperty("results")
```
Comment (fabianschmauder) [inline backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsArtistSearchResults.java:18]: you don't need the json property here

File: backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsMasterReleaseSearchResults.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+@Builder
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class DiscogsMasterReleaseSearchResults {
+
+    @JsonProperty("pagination")
+    DiscogsPagination pagination;
+    @JsonProperty("results")
+    List<DiscogsMasterRelease> results;
```
Comment (fabianschmauder) [inline backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsMasterReleaseSearchResults.java:19]:
```suggestion
    private DiscogsPagination pagination;
    private List<DiscogsMasterRelease> results;
```

File: backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsPagination.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+public class DiscogsPagination {
+
+    @JsonProperty("items")
+    int numberOfItems;
+    @JsonProperty("page")
+    int page;
+    @JsonProperty("pages")
+    int pages;
```
Comment (fabianschmauder) [inline backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsPagination.java:20]:
```suggestion
    @JsonProperty("items")
   private int numberOfItems;
    @JsonProperty("page")
   private int page;
    @JsonProperty("pages")
   private int pages;
```

File: backend/src/main/java/com/hackner/musicompass/discogsapi/service/DiscogsArtistApiService.java [hunk 5]
```diff
@@ -19,23 +23,56 @@
...
+    public List<DiscogsMasterRelease> getDiscogsMasterReleaseListByArtistName(String artistName) {
+
+        try {
+            ResponseEntity<DiscogsMasterReleaseSearchResults> response =
+                    restTemplate.exchange(dynamicUrl(artistName, 1),
+                            HttpMethod.GET,
+                            discogsApiEntityService.createEntity(),
+                            DiscogsMasterReleaseSearchResults.class);
+
+            List<DiscogsMasterRelease> releaseList = response.getBody().getResults();
```
Comment (fabianschmauder) [inline backend/src/main/java/com/hackner/musicompass/discogsapi/service/DiscogsArtistApiService.java:54]: i would move the request to a separate mehtod with page parameter. Than you can use it here and inside of the for loop

File: backend/src/main/java/com/hackner/musicompass/service/ArtistSearchService.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                            .artistInfo(new ArtistInfo().builder()
+                                    .artistImageUrl(discogsArtist.getArtistImageUrl())
+                                    .discogsArtistId(discogsArtist.getDiscogsArtistId())
+                                    .discogsArtistUrl(discogsArtist.getDiscogsArtistUrl()).build())
+                            .build();
+                    return artist;
+                })
+                .collect(Collectors.toList());
+
+        return artistList;
```
Comment (fabianschmauder) [inline backend/src/main/java/com/hackner/musicompass/service/ArtistSearchService.java:43]:
```suggestion
        return discogsArtistList.stream()
                .map(discogsArtist -> new Artist().builder()
                            .artistName(discogsArtist.getArtistName())
                            .artistInfo(new ArtistInfo().builder()
                                    .artistImageUrl(discogsArtist.getArtistImageUrl())
                                    .discogsArtistId(discogsArtist.getDiscogsArtistId())
                                    .discogsArtistUrl(discogsArtist.getDiscogsArtistUrl()).build())
                            .build())
                .collect(Collectors.toList());
```

Comment (fabianschmauder) [reply backend/src/main/java/com/hackner/musicompass/service/ArtistSearchService.java:43]: you could also move 
```
new Artist().builder() .artistName(discogsArtist.getArtistName())
                            .artistInfo(new ArtistInfo().builder()
                                    .artistImageUrl(discogsArtist.getArtistImageUrl())
                                    .discogsArtistId(discogsArtist.getDiscogsArtistId())
                                    .discogsArtistUrl(discogsArtist.getDiscogsArtistUrl()).build())
                            .build()
```

to a map Artist method

File: backend/src/test/java/com/hackner/musicompass/service/ArtistSearchServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+                        .build())
+                .build();
+        Artist artist3 = Artist.builder()
+                .artistName("Hans Bohrer")
+                .artistInfo(new ArtistInfo().builder()
+                        .artistImageUrl("BohrerImageUrl")
+                        .discogsArtistId("333")
+                        .discogsArtistUrl("BohrerUrl")
+                        .build())
+                .build();
```
Comment (fabianschmauder) [inline backend/src/test/java/com/hackner/musicompass/service/ArtistSearchServiceTest.java:87]: you could also move this to a helper method that returns a list of test artists and use it in both tests

**Review by fabianschmauder — COMMENTED** [review]
LGTM

File: backend/src/main/java/com/hackner/musicompass/controller/ArtistSearchController.java (hunk 1 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/controller/ArtistSearchController.java:20]: done. thx

File: backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsArtistSearchResults.java (hunk 2 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsArtistSearchResults.java:18]: I see. I will keep it like this, if sth will be changed there...

File: backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsMasterReleaseSearchResults.java (hunk 3 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsMasterReleaseSearchResults.java:19]: same

File: backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsPagination.java (hunk 4 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/discogsapi/model/DiscogsPagination.java:20]: thx done

File: backend/src/main/java/com/hackner/musicompass/discogsapi/service/DiscogsArtistApiService.java (hunk 5 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/discogsapi/service/DiscogsArtistApiService.java:54]: good point. done

File: backend/src/main/java/com/hackner/musicompass/service/ArtistSearchService.java (hunk 6 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/service/ArtistSearchService.java:43]: I will use the same method also in my second service class (ArtistService.java). Is it fine to create the method here and then import it in the second class? Should I rather create a class "ArtistHelper.java" just to create this method? What is elegantbest practice to use one method in two service classes?

**Pull request merged**